async_transport.go

```diff
@@ -27,6 +28,7 @@ type AsyncTransport struct {
 	PrintPayloadOnError bool
 	bodyChannel         chan payload
 	waitGroup           sync.WaitGroup
+	httpClient          *http.Client
```

Comment:
Btw, wdyt about extracting duplicated code into BaseTransport struct and embedding it in both transports?